compiler-plugin/src/main/kotlin/arrow/meta/quotes/expression/IfExpression.kt

```diff
@@ -17,22 +17,25 @@ import org.jetbrains.kotlin.psi.KtIfExpression
  * import arrow.meta.quotes.ifExpression
  *
  * val Meta.reformatIf: Plugin
- *  get() =
- *  "ReformatIf" {
- *   meta(
- *    ifExpression({ true }) { e ->
- *     Transform.replace(
- *      replacing = e,
- *      newDeclaration = """if ($condition) $`else`""".`if`
- *     )
- *    }
- *   )
- *  }
+ *    get() =
+ *      "Reformat If Expression" {
+ *        meta(
+ *          ifExpression({ true }) { expression ->
+ *            Transform.replace(
+ *              replacing = expression,
+ *              newDeclaration = when {
+ *                `else`.value == null -> """if ($condition) $then""".`if`
+ *                else -> """if ($condition) $then else $`else`""".`if`
+ *              }
+ *            )
+ *          }
+ *        )
+ *      }
  *```
  */
 class IfExpression(
-  override val value: KtIfExpression?,
-  val condition: Scope<KtExpression> = Scope(value?.condition),
-  val then: Scope<KtExpression> = Scope(value),
-  val `else`: Scope<KtExpression> = Scope(value?.`else`)
+  override val value: KtIfExpression,
```

Comment:
I don't think we should let scopes value property as non nullables, if we need to provide in somewhere a nullable `KtIfExpression` reference we will need to let the scope itself as nullable like: `Scope<KtIfExpression>?`, this breaks our scope contract and we have nullable values handling in its `toString` function

Reply:
@bloderxd I think that sounds good - let's definitely visit this because I have some thoughts as well - the only thing holding us back from this choice is depending on the `PsiFactory`, but I think as we move away from that we will have the ability to make better choices like how you suggested!